server/chat/backend/agent/tools/auth/gcp_cached_auth.py

```
@@ -130,17 +130,21 @@ def clear_gcp_cache_for_user(user_id: str) -> None:
-        if stale_path and os.path.exists(stale_path):
-            try:
-                os.remove(stale_path)
-            except Exception as e:
-                logger.debug(f"Could not remove cached SA ADC file {stale_path}: {e}")
+        stale_keys = [k for k in _sa_adc_file_cache if isinstance(k, tuple) and k and k[0] == user_id]
+        for key in stale_keys:
+            stale_path = _sa_adc_file_cache.pop(key, None)
+            if stale_path and os.path.exists(stale_path):
+                try:
+                    os.remove(stale_path)
+                except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)